src/cloud/utils/reporting.ts

```diff
@@ -184,3 +184,26 @@ export const useLoadTimeReporting = (measurement: string) => {
     })
   }, [measurement, loadStartTime])
 }
+
+export const getExperimentVariantId = (
+  experimentID: string,
+  activationEvent: string = 'optimize.activate'
+): string => {
+  const activateOptimize = async (activationEvent: string) => {
+    if (window.dataLayer) {
+      await window.dataLayer.push({event: activationEvent})
+    }
+  }
+  activateOptimize(activationEvent)
+
+  const getVariant = (): string => {
```

Comment:
any reason not to move the definition `activateOptimize` and `getVariant` to outside of `getExperimentVariantId`? right now, every time this function is called, those functions are being re-created then called. looks to me like they only need to be created once